src/t8_schemes/t8_standalone/t8_standalone_implementation.hxx

```diff
@@ -0,0 +1,1568 @@
+        t8_standalone_element<TEclass> child;
+        t8_standalone_scheme<TEclass>::element_get_child ((const t8_element_t *) el, childindex,
+                                                          (t8_element_t *) &child);
+        num_descendants_of_child = t8_standalone_scheme<TEclass>::element_count_leaves ((t8_element_t *) &child, level);
+
+        /* Add number of descendant of current child to cumulative sum */
+        t8_linearidx_t sum_descendants_of_children_until_current
+          = sum_descendants_of_children_before + num_descendants_of_child;
+
+        if (sum_descendants_of_children_until_current > id) {
+          break;
+        }
+
+        /* Go to the next child */
+        sum_descendants_of_children_before = sum_descendants_of_children_until_current;
+        childindex++;
+        T8_ASSERT (childindex < t8_standalone_scheme<TEclass>::element_get_num_children ((const t8_element_t *) el));
+      }
+      /* Replace el by child to go into next iteration at finer level*/
+      t8_standalone_scheme<TEclass>::element_get_child ((const t8_element_t *) el, childindex, (t8_element_t *) el);
+      /* get id in subtree of child */
+      id -= sum_descendants_of_children_before;
+    }
+    T8_ASSERT (id < T8_ELEMENT_NUM_CHILDREN[TEclass]);
+    t8_standalone_scheme<TEclass>::element_get_child ((const t8_element_t *) el, id, (t8_element_t *) el);
+    return;
+  }
+
+  /** Compute the linear id of a given element in a hypothetical uniform
+   * refinement of a given level.
+   * \param [in] elem     The element whose id we compute.
+   * \param [in] level    The level of the uniform refinement to consider.
+   * \return              The linear id of the element.
+   */
+  static inline t8_linearidx_t
+  element_get_linear_id (const t8_element_t *elem, int level)
+  {
+
+    const t8_standalone_element<TEclass> *el = (const t8_standalone_element<TEclass> *) elem;
+    t8_standalone_element<TEclass> ancestor;
+
+    /* Determine the starting element for the iterative linear id computation. */
+    if (level < el->level) {
+      /* Throw away child ids up to the coarser level */
+      t8_standalone_scheme<TEclass>::element_get_ancestor (el, level, &ancestor);
+    }
+    else {
+      /* Start with the input element. 
+       Copy to have a mutable element. */
+      t8_standalone_scheme<TEclass>::element_copy ((const t8_element_t *) el, (t8_element_t *) &ancestor);
+    }
+
+    int id = 0;
+
+    while (ancestor.level != 0) {
+      const int childid = t8_standalone_scheme<TEclass>::element_get_child_id ((t8_element_t *) &ancestor);
+      t8_standalone_scheme<TEclass>::element_get_parent ((t8_element_t *) &ancestor, (t8_element_t *) &ancestor);
+      t8_linearidx_t parent_id = 0;
+
+      for (int ichild = 0; ichild < childid; ichild++) {
+        /* el is now parent, so compute child to get sibling of previous el */
+        t8_standalone_element<TEclass> child;
```

Comment:
Please allocate child outside of both loops. Otherwise we allocate an element a lot of times, even though we only use it as a placeholder.